lib/elements/satin_column.py

```diff
@@ -1066,12 +1072,33 @@ def do_e_stitch(self):
             self.random_width_decrease.any() or self.random_width_increase.any() or self.random_zigzag_spacing,
         )
 
+        short_pairs = self.inset_short_stitches_sawtooth(pairs)
+        max_stitch_length = self.max_stitch_length_px
+        length_sigma = self.random_split_jitter
+        random_phase = self.random_split_phase
+        min_split_length = self.min_random_split_length_px
+        seed = self.random_seed
+        last_point = None
         # "left" and "right" here are kind of arbitrary designations meaning
         # a point from the first and second rail respectively
-        for left, right in pairs:
-            patch.add_stitch(left)
-            patch.add_stitch(right)
-            patch.add_stitch(left)
+        for i, (left, right), (a_short, b_short) in zip(itertools.count(0), pairs, short_pairs):
+            split_points, _ = self.get_split_points(
```

Comment:
This is a good place to add a line `check_stop_flag()` -- I accidentally got my simulator stuck by briefly setting the zigzag spacing to 0 (oops).